### PR TITLE
Adding new image sensor support (IMX585/IMX283)

### PR DESCRIPTION
**Pull request by will127534**
Hi everyone,

I want to add the following sensor support for:

1. IMX283  
    With driver from [https://github.com/will127534/imx283-v4l2-driver](https://github.com/will127534/imx283-v4l2-driver) + Tuning files 
2. IMX585
    With driver from [https://github.com/will127534/imx585-v4l2-driver](https://github.com/will127534/imx585-v4l2-driver) + Tuning files

### CODE REVIEW

**Comment by kbingham** [comment]
Hi Will,

Patch 1 should be possible to accept now, but as it's for RPi needs a review tag from David/Naush.
Patch 2 is still blocked on the IMX585 driver not yet being upstream. 
Patch 3 could be merged with a Signed-off-by: tag added and review, but that should be easy on that one.

**Comment by naushir** [comment]
@will127534 can you make the changes suggested above when you get a change.  Once that's done, I can create a git pull request for @kbingham to merge into the main libcamera tree.  Then we will backport into the RPi tree.

**Comment by will127534** [comment]
Hi @kbingham and @naushir,
I've updated the patch 3 to include the signoff.
For IMX585 I'm/will be working to upstream the kernel driver, is it possible to merge the IMX283 patch first?

The reason I asked is because for some reason I can't send a pull request based on a specific git commit but HEAD only....

**Comment by naushir** [comment]
I'll submit these changes to the mailing list today.  @will127534 I will make slight edits to the commit message to match the libcamera guidelines.  Hope that's ok.

File: src/ipa/rpi/cam_helper/cam_helper_imx585.cpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	/*
+	 * Smallest difference between the frame length and integration time,
+	 * in units of lines.
+	 */
+	static constexpr int frameIntegrationDiff = 4;
+};
+
+/*
+ * Imx585 doesn't output metadata, so we have to use the "unicam parser" which
+ * works by counting frames.
```
Comment (naushir) [inline src/ipa/rpi/cam_helper/cam_helper_imx585.cpp:34]: "unicam parser" should be replaced by "delayed controls" here.

File: src/ipa/rpi/vc4/data/imx585.json [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+                    "daylight":
+                    {
+                        "lo": 5500,
+                        "hi": 6500
+                    },
+                    "cloudy":
+                    {
+                        "lo": 7000,
+                        "hi": 8600
+                    }
```
Comment (naushir) [inline src/ipa/rpi/vc4/data/imx585.json:112]: We should remove the cloudy mode as the ct curve below ends at 6500k, so this mode is out of range.

**Review by naushir — COMMENTED** [review]
Minor changes needed + some whitespaces fixes in the .cpp file. I'll make these changes and post this to the libcamera list.

**Pull request closed without merging**